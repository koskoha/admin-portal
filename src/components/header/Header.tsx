import * as React from 'react';
import { styled } from 'baseui';
import { Link } from 'react-router-dom';

import logoIcon from '../../images/logo.png';

const HeaderWrapper = styled('div', {
	marginLeft: 'auto',
	marginRight: 'auto',
	display: 'grid',
	gridGap: '30px',
	gridTemplateColumns: '40px auto',
	maxWidth: '1300px',
	paddingLeft: '20px',
	paddingRight: '20px'
});

const HeaderTitle = styled('div', {
	paddingTop: '20px',
	display: 'grid',
	gridTemplateColumns: 'auto auto'
});

const UserMenu = styled('div', {
	display: 'flex',
	flexDirection: 'row',
	justifyContent: 'flex-end'
});

const HeaderBlock = styled('div', {
	width: '100%',
	backgroundColor: '#00579b',
	color: '#fff'
});

const A = styled(Link, {
	marginLeft: '30px',
	display: 'block',
	color: '#fff',
	fontSize: '18px',
	fontWeight: 'bold',
	textDecoration: 'none'
});

const P = styled('p', {
	fontFamily: 'Lato-Regular',
	fontSize: '14px',
	color: '#fff',
	textAlign: 'right',
	letterSpacing: '0.02em',
	lineHeight: '24px'
});

class Header extends React.Component {
	render() {
		return (
			<HeaderBlock>
				<HeaderWrapper>
					<HeaderTitle>
						<div>
							<img src={logoIcon} alt="logo" />
						</div>
						<div className="company-title">
							<div className="title">SpecialtyCare Connect</div>
							<div className="sub-title">POWERED BY ARMADAHEALTH</div>
						</div>
					</HeaderTitle>
					<div>
						<P>FirstName LastName</P>
						<UserMenu>
							<A to="#">My Profile</A>
							<A to="#">Log Out</A>
						</UserMenu>
					</div>
				</HeaderWrapper>
			</HeaderBlock>
		);
	}
}

export default Header;
