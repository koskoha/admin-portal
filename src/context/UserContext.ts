import * as React from 'react';

const UserContext = React.createContext({ authenticated: false, x: 0 });

export default UserContext;
