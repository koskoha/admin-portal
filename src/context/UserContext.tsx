import * as React from 'react';

import { AUTH_TOKEN } from '../config/constant';
import { isTokenExpired } from '../helper/jwtHelper';

interface UserContextInterface {
	token: string | undefined | null;
	logout: () => void;
	refreshTokenFn: () => void;
}

export const UserContext = React.createContext<UserContextInterface>({
	token: undefined,
	logout: () => {},
	refreshTokenFn: () => {}
});

interface UserProviderProps {
	children: React.ReactNode;
	token: string | null;
}

interface UserProviderInterProps {
	token: string | undefined | null;
}

export class UserProvider extends React.Component<
	UserProviderProps,
	UserProviderInterProps
> {
	constructor(props) {
		super(props);
		this.state = {
			token: props.token
		};
	}

	bootStrapData() {
		try {
			const token = localStorage.getItem(AUTH_TOKEN);
			if (token !== null && token !== undefined) {
				const expired = isTokenExpired(token);
				if (!expired) {
					this.setState({ token });
				} else {
					localStorage.removeItem(AUTH_TOKEN);
					this.setState({ token: undefined });
				}
			}
		} catch (e) {
			console.log(e);
		}
	}

	logout = () => {
		localStorage.removeItem(AUTH_TOKEN);
		this.setState({ token: undefined });
	};

	// verify localStorage check
	componentDidMount() {
		this.bootStrapData();
	}

	refreshTokenFn = (data = {}) => {
		const token = data[AUTH_TOKEN];

		if (token) {
			localStorage.setItem(AUTH_TOKEN, token);
		} else {
			localStorage.removeItem(AUTH_TOKEN);
		}

		this.setState({
			token: data[AUTH_TOKEN]
		});
	};

	render() {
		const { children } = this.props;
		const { token } = this.state;
		return (
			<UserContext.Provider
				value={{
					token,
					logout: this.logout,
					refreshTokenFn: this.refreshTokenFn
				}}
			>
				{children}
			</UserContext.Provider>
		);
	}
}
