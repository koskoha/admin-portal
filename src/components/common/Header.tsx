import * as React from 'react';
import { Link } from 'react-router-dom';

import logoIcon from '../../images/logo.png';
import { UserContext } from '../../context/UserContext';

const Header: React.FC = () => {
	const { logout } = React.useContext(UserContext);
	return (
		<div className="header">
			<div className="header-wrapper">
				<div className="header-title">
					<div>
						<img src={logoIcon} alt="logo" />
					</div>
					<div className="company-title">
						<div className="title">SpecialtyCare Connect</div>
						<div className="sub-title">POWERED BY ARMADAHEALTH</div>
					</div>
				</div>
				<div>
					<p className="admin-name">FirstName LastName</p>
					<div className="user-menu">
						<button className="menu-link menu-btn" onClick={() => logout()}>
							Log Out
						</button>
						<Link className="menu-link" to="#">
							My Profile
						</Link>
					</div>
				</div>
			</div>
		</div>
	);
};

export default Header;
