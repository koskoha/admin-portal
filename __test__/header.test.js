import * as React from 'react';
import '@babel/polyfill';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';

import Header from '../src/components/common/Header';
import { UserProvider } from '../src/context/UserContext';

describe('Tests for <Header /> component', () => {
	it('should render without error', () => {
		mount(
			<Router>
				<Header />
			</Router>
		);
	});

	it('should match snapshot', () => {
		const wrapper = mount(
			<UserProvider>
				<Router>
					<Header />
				</Router>
			</UserProvider>
		);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	it('call logout function on "Log out" btn click', () => {
		const wrapper = mount(
			<UserProvider>
				<Router>
					<Header />
				</Router>
			</UserProvider>
		);
		const component = wrapper.find(Header).instance();
		const spy = jest.spyOn(component, 'logout');

		wrapper.find('button[role="logout"]').simulate('click');
		wrapper.update();

		expect(spy).toHaveBeenCalled();
		expect(spy).toHaveBeenCalledTimes(1);
	});
});
