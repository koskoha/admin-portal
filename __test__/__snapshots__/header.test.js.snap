// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Header /> component should match snapshot 1`] = `
<UserProvider>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Header>
        <div
          className="header"
        >
          <div
            className="header-wrapper"
          >
            <div
              className="header-title"
            >
              <div>
                <img
                  alt="logo"
                  src="test-file-stub"
                />
              </div>
              <div
                className="company-title"
              >
                <div
                  className="title"
                >
                  SpecialtyCare Connect
                </div>
                <div
                  className="sub-title"
                >
                  POWERED BY ARMADAHEALTH
                </div>
              </div>
            </div>
            <div>
              <p
                className="admin-name"
              >
                FirstName LastName
              </p>
              <div
                className="user-menu"
              >
                <button
                  className="menu-link menu-btn"
                  onClick={[Function]}
                  role="logout"
                >
                  Log Out
                </button>
                <Link
                  className="menu-link"
                  to="#"
                >
                  <a
                    className="menu-link"
                    href="/"
                    onClick={[Function]}
                  >
                    My Profile
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </Header>
    </Router>
  </BrowserRouter>
</UserProvider>
`;
